Mathlib/GroupTheory/Perm/Cycle/Factors.lean

```diff
@@ -42,19 +35,101 @@ section CycleOf
 
 variable [DecidableEq α] [Fintype α] {f g : Perm α} {x y : α}
 
-/-- `f.cycleOf x` is the cycle of the permutation `f` to which `x` belongs. -/
-def cycleOf (f : Perm α) (x : α) : Perm α :=
+/- This section generalizes a `(f : Perm α) → DecidableRel (SameCycle f)` instance in some
+declarations to make `cycleOf` computable.
+All declarations in this section are specialized later.
+-/
+section GeneralizedDecRel
+
+variable [(f : Perm α) → DecidableRel (SameCycle f)]
```

Comment:
What was the reason for asking for this and not the less demanding ` [DecidableRel (SameCycle f)]`?

Reply:
When `f : Perm α` appears as an explicit paramater after this instance, the error occurs:
```lean
variable [DecidableRel (SameCycle f)]

private def cycleOfGDR (f : Perm α) (x : α) : Perm α :=
  ofSubtype (subtypePerm f fun _ => sameCycle_apply_right.symm : Perm { y // SameCycle f x y })
-- failed to synthesize instance
--   DecidablePred fun x_1 ↦ f.SameCycle x x_1
```

Reply:
After all, any declarations in this section aren't used in computations so it's not a big issue.